spec/patterns/aws_spec.rb

```diff
@@ -0,0 +1,393 @@
+        should include("proto" => "http" )
+        should include("httpversion" => "1.1" )
+        should include("urihost" => "www.example.com:80" )
+        should include("path" => "/" )
+      end
+    end
+
+    ["tags", "params"].each do |attribute|
+      it "have #{attribute} as nil" do
+        expect(subject[attribute]).to be_nil
+      end
+    end
+  end
+
+  context "parsing a PUT request access log with missing backend info" do
+
+    let(:message) do
+      '2015-04-10T08:11:09.865823Z us-west-1-production-media 49.150.87.133:55128 - -1 -1 -1 408 - 1294336 0 "PUT https://media.xxxyyyzzz.com:443/videos/F4_M-T4X0MM6Hvy1PFHesw HTTP/1.1"'
+    end
+
+    it "matches" do
+      expect(grok).to include("timestamp"=>"2015-04-10T08:11:09.865823Z")
+      if ecs_compatibility?
+        expect(grok).to include("url"=>{
+            "original"=>"https://media.xxxyyyzzz.com:443/videos/F4_M-T4X0MM6Hvy1PFHesw",
+            "scheme"=>"https", "port"=>443, "path"=>"/videos/F4_M-T4X0MM6Hvy1PFHesw", "domain"=>"media.xxxyyyzzz.com"
+        })
+        expect(grok).to include("source"=>{"port"=>55128, "ip"=>"49.150.87.133"})
+        expect(grok).to include("http"=>{
+            "request"=>{"method"=>"PUT", "body"=>{"bytes"=>1294336}}, "version"=>"1.1",
+            "response"=>{"body"=>{"bytes"=>0}, "status_code"=>408}
+        })
+        # no backend.ip and backend.port
+        # no backend.http.status.code
+        # no request_processing_time.sec and friends
+        expect(grok).to include("aws"=>{"elb"=>{"name"=>"us-west-1-production-media"}})
+      else
+        expect(grok).to include(
+            "elb"=>"us-west-1-production-media",
+            "clientip"=>"49.150.87.133", "clientport"=>55128,
+            "response_processing_time"=>-1.0,
+            "request_processing_time"=>-1.0,
+            "backend_processing_time"=>-1.0,
+            "response"=>408,
+            "received_bytes"=>1294336,
+            "bytes"=>0,
+            "verb"=>"PUT",
+            "request"=>"https://media.xxxyyyzzz.com:443/videos/F4_M-T4X0MM6Hvy1PFHesw",
+            "port"=>"443", "proto"=>"https", "path"=>"/videos/F4_M-T4X0MM6Hvy1PFHesw", "urihost"=>"media.xxxyyyzzz.com:443",
+            "httpversion"=>"1.1")
+
+        expect(grok.keys).to_not include("backendip", "backendport", "backendresponse")
+      end
+    end
+
+  end
+
+  context '(new) https format' do # slightly longer - 3 fields added at the end
+
+    let(:message) do
+      '2015-05-13T23:39:43.945958Z my-loadbalancer 192.168.131.39:2817 10.0.0.1:80 0.000086 0.001048 0.001337 200 200 0 57 "GET https://www.example.com:443/ HTTP/1.1" "curl/7.38.0 (#56-0ef1d4a5)" DHE-RSA-AES128-SHA TLSv1.2'
+    end
+
+    it 'matches (new) suffix fields' do
+      if ecs_compatibility?
+        expect(grok).to include "tls" => { "cipher" => "DHE-RSA-AES128-SHA" }
+        expect(grok).to include "aws" => { "elb" => hash_including("ssl_protocol" => 'TLSv1.2')}
```

Comment:
we match `tls.cipher` directly instead of a `aws.elb.ssl_cipher` (Beats starts with the `aws.elb` version),
but capture `aws.elb.ssl_protocol` as it needs (lowercase) normalization to capture as `tls.version` ... we could attempt to parse out the `tls.version` from the `ssl_protocol` field as part of the matching - let me know if it feels like we should do that, at least.